File: Paint_FrontEnd/src/app/Shapes/ellipse.ts
import { IShape } from "./ishape";

export class Ellipse extends IShape{
    private _radiusX: number;

    private _radiusY: number;

    constructor(x: number, y: number, stroke: string, strokeWidth: number, fill: string, rotate: number, draggable: boolean, id: string, update: boolean, deleteflag: boolean, hRadius: number, vRadius: number){
        super(x, y, stroke, strokeWidth, fill, rotate, draggable, id, update, deleteflag);
        this._radiusX = hRadius;
        this._radiusY = vRadius;
        this.type = "ellipse";
    }

    public get hRadius(): number {
        return this._radiusX;
    }
    public set hRadius(value: number) {
        this._radiusX = value;
    }
    public get vRadius(): number {
        return this._radiusY;
    }
    public set vRadius(value: number) {
        this._radiusY = value;
    }

}

File: Paint_BackEnd/src/main/java/com/paint/Paint_BackEnd/Shapes/Ellipse.java
package com.paint.paint_backend.Shapes;

public class Ellipse extends IShape{
    private int _radiusX;
    private int _radiusY;
    Ellipse(int x, int y, String stroke, double strokeWidth, String fill, int rotate, boolean draggable, String id, boolean update, boolean deleteflag, int radiusX, int radiusY){
        super(x, y, stroke, strokeWidth, fill, rotate, draggable, id, update, deleteflag);
        this._radiusX = radiusX;
        this._radiusY = radiusY;
        super.set_type("ellipse");
    }


    public int get_radiusX() {
        return _radiusX;
    }

    public void set_radiusX(int _radiusX) {
        this._radiusX = _radiusX;
    }

    public int get_radiusY() {
        return _radiusY;
    }

    public void set_radiusY(int _radiusY) {
        this._radiusY = _radiusY;
    }
}
